agent-inject/agent/container_env.go

```diff
@@ -64,6 +65,48 @@ func (a *Agent) ContainerEnvVars(init bool) ([]corev1.EnvVar, error) {
 			Name:  "VAULT_CONFIG",
 			Value: b64Config,
 		})
+	} else {
+		// set up environment variables to access Vault since "vault" section may not be present in the config
+		if a.Vault.Address != "" {
+			envs = append(envs, corev1.EnvVar{
+				Name:  "VAULT_ADDR",
+				Value: a.Vault.Address,
+			})
+		}
```

Comment:
(Just making sure I understand) It looks like all these can be overridden by the user with annotations from https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-annotations? So like for the vault address, the order of precedence would be:
* `vault` stanza in configmap
* This setting of VAULT_ADDR
* The value of `vault.hashicorp.com/service`

We may want to add a note about this new behavior in or adjacent to https://www.vaultproject.io/docs/platform/k8s/injector/examples

Reply:
Yes. I believe that is correct. I'll double-check it before merging, and add a ticket to update those docs.

Reply:
Yes, I confirmed that `vault.hashicorp.com/service` will override the agent container `VAULT_ADDR` settings.